### PR TITLE
fix product image copy issue #3932

### PR DESCRIPTION
**Pull request by deepaksinghgusain227**
**BUGS: Fixed**

Fixed issued #3932 

Previously
When Admin want to copy the product then the image section not copy to new directory instead it link to that product id 
which the product you copied

Now
when admin wants to copy a product then the image section make a new copy and image get uploaded to that new directory.
No link to previous product id

### CODE REVIEW

**Comment by vaishaliwebkul** [comment]
Working fine

File: packages/Webkul/Product/src/Repositories/ProductRepository.php [hunk 1]
```diff
@@ -122,9 +122,9 @@ public function getAll($categoryId = null)
         if (core()->getConfigData('catalog.products.storefront.products_per_page')) {
             $pages = explode(',', core()->getConfigData('catalog.products.storefront.products_per_page'));
 
-            $perPage = isset($params['limit']) ? (! empty($params['limit']) ? $params['limit'] : 9) : current($pages);
```
Comment (jitendra-webkul) [inline packages/Webkul/Product/src/Repositories/ProductRepository.php:125]: Always use space after ! operator

**Comment by rahulcs0082** [comment]
Hi @deepaksinghgusain227 

The same mistake is done many times in the given file, changed all of them.  (https://prnt.sc/upqj6h)

File: packages/Webkul/Product/src/Repositories/ProductRepository.php [hunk 2]
```diff
@@ -175,14 +175,14 @@ public function getAll($categoryId = null)
...
-                $orderDirection = ! empty($sortOptions) ? $sortOptions[1] : 'asc';
+                $orderDirection = !empty($sortOptions) ? $sortOptions[1] : 'asc';
             }
 
             if (isset($params['sort'])) {
                 $this->checkSortAttributeAndGenerateQuery($qb, $params['sort'], $orderDirection);
             } else {
                 $sortOptions = $this->getDefaultSortByOption();
-                if (! empty($sortOptions)) {
+                if (!empty($sortOptions)) {
```
Comment (devansh-webkul) [inline packages/Webkul/Product/src/Repositories/ProductRepository.php:185]: Well, I guess your changes again changed. Please check these in all codes.

File: packages/Webkul/Product/src/Repositories/ProductRepository.php [hunk 3]
```diff
@@ -684,37 +683,48 @@ private function persistRelations($originalProduct, $copiedProduct): void
...
+               
+                $imagePath = $image['path'];
+                $imagename = explode('/', $imagePath);
+                $dir = 'product/' . $copiedProduct['id'];
+
+                Storage::copy($imagePath, $dir . '/' . $imagename[2]);
+                
+                $image['product_id'] =  $copiedProduct['id'];
+                $image['path'] = $dir . '/' . $imagename[2];
+
```
Comment (devansh-webkul) [inline packages/Webkul/Product/src/Repositories/ProductRepository.php:709]: Please make a method for this.

**Review by devansh-webkul — COMMENTED** [review]
Please check all the PSR 2 guidelines. Maybe your all changes get reverted.

**Comment by deepaksinghgusain227** [comment]
closed because new pr created

**Pull request closed without merging**